### PR TITLE
0.12.0 release

### PR DESCRIPTION
**Pull request by dstebila**
I've updated the release notes for the 0.12.0 release with reference to the CVE for HQC. 

Once this PR is approved, I will merge and then tag the release.

### CODE REVIEW

File: RELEASE.md
Comment (SWilson4) [inline RELEASE.md:1]: The "Full Changelog" link still points to the 0.12.0-rc1 tag. I can't leave a suggestion on that part of the file, but all that should be required is updating the tag and adding a changelog line for PR #2006 (if desired).

I also just noticed that commits cce1bfde4e52c524b087b9687020d283fbde0f24 and 95f904bcaafe4cb12cfdf029347ac3b7c795308a don't appear in the auto-generated changelog, since they didn't come from a PR—worth manually adding a line for them?

**Comment by SWilson4** [comment]
[Extended tests](https://github.com/open-quantum-safe/oqs-provider/actions/runs/12202027558/job/34041770578) and [oqs-provider release tests](https://github.com/open-quantum-safe/oqs-provider/actions/runs/12202137355) have been triggered. The latter run is using the [ds-0.12.0-tracker](https://github.com/open-quantum-safe/oqs-provider/tree/ds-0.12.0-tracker) branch, which I just created from the branch for open-quantum-safe/oqs-provider#568.

File: RELEASE.md
Comment (dstebila) [reply RELEASE.md:1]: Thanks Spencer. I've updated the release notes to address this.

**Review by SWilson4 — APPROVED** [review]

**Comment by dstebila** [comment]
Waiting for a second review from @open-quantum-safe/liboqs-committers

**Pull request merged**
